### PR TITLE
Update: Add notification for when a new preview is available

### PR DESCRIPTION
**Pull request by priyajeet**
* Also adds a prepush script to build the project and add locales to prevent any missing locales in commit
* Adds a couple of missing locales

### CODE REVIEW

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #505

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/components/ContentPreview/ContentPreview.js [hunk 1]
```diff
@@ -612,7 +623,27 @@ class ContentPreview extends PureComponent<Props, State> {
...
+        // In this case preview should reload without prompting the user
+        if (isWatermarked || !isExistingFile) {
+            this.setState({ file, isFileError: false, isReloadNotificationVisible: false });
+            return;
+        }
+
+        // If we are already prevewing the file that got updated then show the
+        // user a notification to reload the file only if its sha1 changed
+        // $FlowFixMe file version and sha1 should exist at this point
+        if (currentFile.file_version.sha1 !== file.file_version.sha1) {
```
Comment (jeremypress) [inline src/components/ContentPreview/ContentPreview.js]: Save this boolean in a variable and then set the state with it's result? Can get rid of the else case.

File: src/components/ContentPreview/ContentPreview.js [hunk 2]
```diff
@@ -950,19 +985,18 @@ class ContentPreview extends PureComponent<Props, State> {
                     )}
                     <div className='bcpr-body'>
                         <div className='bcpr-container' onMouseMove={this.onMouseMove} ref={this.containerRef}>
-                            {!file ? (
+                            {file ? (
```
Comment (jeremypress) [inline src/components/ContentPreview/ContentPreview.js]: 👍

File: src/api/__tests__/File-test.js [hunk 3]
```diff
@@ -349,21 +378,22 @@ describe('api/File', () => {
...
+                get: jest.fn().mockReturnValueOnce(Promise.resolve({ data: 'new file' }))
             };
 
             const success = jest.fn();
 
-            return file.file('id', success, 'error', true).then(() => {
+            return file.getFile('id', success, 'error', { forceFetch: false, refreshCache: true }).then(() => {
                 expect(file.getCacheKey).toHaveBeenCalledWith('id');
-                expect(success).toHaveBeenCalledWith({ file: 'new file' });
+                expect(success.mock.calls.length).toBe(2);
```
Comment (DanDeMicco) [inline src/api/__tests__/File-test.js]: You can use ```.toHaveBeenCalledTimes(number)```

File: src/api/__tests__/File-test.js [hunk 4]
```diff
@@ -349,21 +378,22 @@ describe('api/File', () => {
...
+                expect(success.mock.calls[0][0]).toBe('file');
```
Comment (DanDeMicco) [inline src/api/__tests__/File-test.js]: Just saw there is ```toHaveBeenNthCalledWith```

File: src/components/ContentPreview/ContentPreview.js [hunk 5]
```diff
@@ -612,7 +623,27 @@ class ContentPreview extends PureComponent<Props, State> {
...
+        // If the file is watermarked or if its a new file, then update the state
+        // In this case preview should reload without prompting the user
+        if (isWatermarked || !isExistingFile) {
+            this.setState({ file, isFileError: false, isReloadNotificationVisible: false });
+            return;
+        }
+
+        // If we are already prevewing the file that got updated then show the
+        // user a notification to reload the file only if its sha1 changed
+        // $FlowFixMe file version and sha1 should exist at this point
```
Comment (DanDeMicco) [inline src/components/ContentPreview/ContentPreview.js]: Make ```file_version``` not be optional in ```BoxItem```?

File: src/components/ContentPreview/ReloadNotification.scss [hunk 6]
```diff
@@ -0,0 +1,9 @@
+@import '../variables';
+
+.bcpr {
+    .bcpr-notification {
+        left: 50%;
```
Comment (DanDeMicco) [inline src/components/ContentPreview/ReloadNotification.scss]: None of the typical horizontal centering stuff works e.g. text-align, margin:auto?

**Pull request merged**